spec/datadog/ci/flush_spec.rb

```diff
@@ -13,14 +13,12 @@
     instance_double(
       Datadog::Tracing::TraceOperation,
       origin: origin,
-      sampled?: sampled,
       finished?: finished,
       flush!: trace
     )
   end
 
   let(:origin) { 'ci-origin' }
-  let(:sampled) { true }
   let(:finished) { true }
 
```

Comment:
Is it just me or is `Datadog::CI::Flush::Tagging` a really bizarre part of the code? E.g. the class that is supposed to decide if something is ready for flushing or not has added behavior to mutate all spans to add an extra tag. Weeeird way of doing it... 👀

Reply:
It seems like it's due to a lack of a clear interception point between the Tracing and CI products.
A CI formatter might make more sense, not sure.